test/Atmos/EDMF/bomex_edmf.jl

```diff
@@ -46,8 +46,13 @@ function init_state_prognostic!(
     # a moist_thermo state is used here to convert the input θ,q_tot to e_int, q_tot profile
     e_int = internal_energy(m, state, aux)
 
-    q_tot = state.moisture.ρq_tot / state.ρ
-    ts = PhaseEquil(m.param_set, e_int, state.ρ, q_tot)
+    if m.moisture isa DryModel
+        ρq_tot = 0
```

Comment:
Is this type stable?

Reply:
We'd have to be aware of promotion rules downstream. I'll wrap it in `FT` so we won't need to think about it.